src/persist/src/nemesis/validator.rs

```diff
@@ -151,9 +162,8 @@ impl Validator {
     }
 
     fn step_restart(&mut self, req_id: ReqId, res: Result<(), Error>) {
-        // At the moment, there are no nemeses that would prevent a restart
-        // from working, so we shouldn't see any errors.
-        let should_succeed = true;
+        let should_succeed = self.storage_available;
         self.check_success(req_id, &res, should_succeed);
+        self.runtime_available = res.is_ok();
```

Comment:
I'm trying to understand the semantics of `runtime_available` -- it seems like we expect restarts to fail when the storage is unavailable, and failed restarts somehow make the runtime unavailable? I guess my confusion is that when the storage is unavailable I expect all operations including restarts to be a no-op when it seems like even a failed restart of an unavailable stream has an affect. A comment indicating why the runtime is unavailable would help here

Reply:
Done!

```
        // The semantics of Req::Restart are pretty blunt. It unconditionally
        // stops the currently running persister and then attempts to start a
        // new one. If the storage is down, the new one won't be able to read
        // metadata and will fail to start. This will cause all operations on
        // the persister to fail until it gets another Req::Restart with storage
        // available. This ends up being a pretty uninteresting state to test,
        // but it's not clear what else we should do. Perhaps we should rework
        // the weights so that StorageAvailable and Restart are both much more
        // likely while the runtime is down.
```

Reply:
got it. I think maybe it would be clearer to break `Restart` into `Stop` and `Start` and make `StorageAvailable` happen more often when the storage is unavailable and `Start` happen more often the runtime is stopped. Although, it might just not be worth the extra complexity.